From 7a5d58bffd20eda99f9d390539ee7c5f41fcb5e8 Mon Sep 17 00:00:00 2001
From: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Date: Mon, 5 Aug 2024 03:33:29 -0700
Subject: [PATCH 909/955] ci: change devargs to support Rx-inject

Rx-inject need nix device and an sso device
Add these devices in the devargs

Signed-off-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Change-Id: I32986aac0e46c635da85038b46e3fa5161fa9ade
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/132804
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Anoob Joseph <anoobj@marvell.com>
---
 .../test/cnxk-tests/crypto_autotest/crypto_autotest.sh    | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/marvell-ci/test/cnxk-tests/crypto_autotest/crypto_autotest.sh b/marvell-ci/test/cnxk-tests/crypto_autotest/crypto_autotest.sh
index f7baa9fbcb8e1..687a285036e1d 100755
--- a/marvell-ci/test/cnxk-tests/crypto_autotest/crypto_autotest.sh
+++ b/marvell-ci/test/cnxk-tests/crypto_autotest/crypto_autotest.sh
@@ -8,8 +8,14 @@ SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
 
 CNXK_MEMPOOL_DEVICE=$(lspci -d :a0fb | tail -1 | awk -e '{ print $1 }')
 
+NIX_INL_DEV=${NIX_INL_DEV:-$(lspci -d :a0f0 | tail -1 | awk -e '{ print $1 }')}
+NIX_INL_DEVICE="$NIX_INL_DEV"
+
+ETH_DEV=${ETH_DEV:-$(lspci -d :a0f8 | head -1 | awk -e '{ print $1 }')}
+ETHERNET_DEVICE="$ETH_DEV"
+
 CN10K_CRYPTO_DEVICE="0002:20:00.1"
-CN10K_EAL_ARGS="-a $CN10K_CRYPTO_DEVICE,max_qps_limit=4 -a $CNXK_MEMPOOL_DEVICE"
+CN10K_EAL_ARGS="-a $ETHERNET_DEVICE -a $NIX_INL_DEVICE,rx_inject_qp=1 -a $CN10K_CRYPTO_DEVICE,max_qps_limit=4,rx_inject_qp=1 -a $CNXK_MEMPOOL_DEVICE"
 CN10K_EAL_ARGS+=" --log-level=7"
 
 CN9K_CRYPTO_DEVICE="0002:10:00.1"
-- 
2.25.1

